src/values/IntegralValue.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import { NumberValue, Value } from "./index.js";
+import type { Realm } from "../realm.js";
+
+export default class IntegralValue extends NumberValue {
+  constructor(realm: Realm, value: number, intrinsicName?: string) {
+    super(realm, value, intrinsicName);
+  }
+
+  equals(x: Value): boolean {
```

Comment:
Unless the constructor for NumberValue is modified to return an instance of IntegralValue when ever a number that happens to be an integer is passed to it, we cannot guarantee that any integral number value will necessarily be an instance of IntegralValue.